Mathlib/NumberTheory/NumberField/ProductFormula.lean

```diff
@@ -0,0 +1,111 @@
+/-
+Copyright (c) 2024 Fabrizio Barroero. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Fabrizio Barroero
+-/
+import Mathlib.NumberTheory.NumberField.FinitePlaces
+
+/-!
+# The Product Formula for number fields
+In this file we prove the Product Formula for number fields: for any non-zero element `x` of a
+number field `K`, we have `∏|x|ᵥ=1` where the product runs over the equivalence classes of absoulte
+values of `K` and the `|⬝|ᵥ` are suitably normalized.
+
+## Main Results
+* `NumberField.FinitePlace.prod_eq_inv_abs_norm`: for any non-zero element `x` of a number field
+`K`, the product `∏|x|ᵥ` of the absolute values of `x` associated to the finite places of `K` is
+equal to the inverse of the norm of `x`.
+* `NumberField.prod_abs_eq_one`: for any non-zero element `x` of a number field `K`, we have
+`∏|x|ᵥ=1` where the product runs over the equivalence classes of absoulte values of `K`.
+
+## Tags
+number field, embeddings, places, infinite places, finite places, product formula
+-/
+
+namespace NumberField
+
+variable {K : Type*} [Field K] [NumberField K]
+
+open Algebra Classical FinitePlace Function Ideal IsDedekindDomain
+  IsDedekindDomain.HeightOneSpectrum
+
+/-- For any non-zero `x` in `𝓞 K`, the prduct of `w x`, where `w` runs over `FinitePlace K`, is
+equal to the inverse of the absolute value of `Algebra.norm ℤ x`. -/
+theorem FinitePlace.prod_eq_inv_abs_norm_int {x : 𝓞 K} (h_x_nezero : x ≠ 0) :
+    ∏ᶠ w : FinitePlace K, w x = (|norm ℤ x| : ℝ)⁻¹ := by
+  convert_to ∏ᶠ v : HeightOneSpectrum (𝓞 K), ‖embedding v x‖ = |↑(norm ℤ x)|⁻¹
+  · exact (finprod_eq_of_bijective maximalIdeal ((bijective_iff_existsUnique _).mpr
+      <| fun v ↦ ⟨mk v, maximalIdeal_mk v, fun _ a ↦ by rw [← a, mk_maximalIdeal]⟩)
+      (fun w ↦ (norm_embedding_eq w (x : K)).symm))
+  apply (inv_eq_of_mul_eq_one_left _).symm
+  norm_cast
+  have h_span_nezero : span {x} ≠ 0 := by
+    simp only [Submodule.zero_eq_bot, ne_eq, span_singleton_eq_bot, h_x_nezero, not_false_eq_true]
```

Comment:
```suggestion
    simp [h_x_nezero]
```
For terminal simps it seems that we're now being encouraged not to squeeze them, because it makes maintaining them harder. Nobody wants to see `not_false_eq_true` anyway :-)